api/src/main/java/org/asynchttpclient/request/body/generator/ReactiveStreamBodyGenerator.java

```diff
@@ -0,0 +1,152 @@
+     */
+    public ReactiveStreamBodyGenerator(Executor executor, Publisher<ByteBuffer> publisher) {
+        this.executor = executor;
+        this.publisher = publisher;
+        this.feedableBodyGenerator = new FeedableBodyGenerator();
+    }
+
+    public Publisher<ByteBuffer> getPublisher() {
+        return this.publisher;
+    }
+
+    @Override
+    public void writeChunkBoundaries() {
+        feedableBodyGenerator.writeChunkBoundaries();
+    }
+
+    @Override
+    public void setListener(FeedListener listener) {
+        feedableBodyGenerator.setListener(listener);
+    }
+
+    @Override
+    public Body createBody() {
+        return new StreamedBody(executor, publisher, feedableBodyGenerator);
+    }
+
+    private class StreamedBody implements Body {
+        private final SimpleSubscriber subscriber;
+        private final Body body;
+
+        public StreamedBody(Executor executor, Publisher<ByteBuffer> publisher, FeedableBodyGenerator bodyGenerator) {
+            if(executor == null)
+                throw new IllegalStateException("An Executor is needed to stream the body. Hint: Create an instance of ReactiveStreamBodyGenerator with a non null Executor.");
+            this.body = bodyGenerator.createBody();
+            this.subscriber = new SimpleSubscriber(executor, bodyGenerator);
+            publisher.subscribe(subscriber);
+        }
+
+        @Override
+        public void close() throws IOException {
+            body.close();
+        }
+
+        @Override
+        public long getContentLength() {
+            return body.getContentLength();
+        }
+
+        @Override
+        public State read(ByteBuffer buffer) throws IOException {
+            return body.read(buffer);
+        }
+    }
+
+    private static class SimpleSubscriber implements Subscriber<ByteBuffer> {
+
+        private static final Logger LOGGER = LoggerFactory.getLogger(SimpleSubscriber.class);
+
+        private final Executor executor;
+        private final FeedableBodyGenerator feeder;
+        private volatile Subscription subscription;
+
+        public SimpleSubscriber(Executor executor, FeedableBodyGenerator feeder) {
+            this.executor = executor;
+            this.feeder = feeder;
+        }
+
+        @Override
+        public void onSubscribe(Subscription s) {
+            if (s == null) throw null;
+
+            // If someone has made a mistake and added this Subscriber multiple times, let's handle it gracefully
+            if (this.subscription != null) { 
+                s.cancel(); // Cancel the additional subscription
+            }
+            else {
+              subscription = s;
+              subscription.request(1);
+            }
+        }
+
+        @Override
+        public void onNext(ByteBuffer t) {
+            if (t == null) throw null;
+            try {
+                feeder.feed(t, false);
+                executor.execute(new Runnable() {
```

Comment:
This is not necessary, reactive streams spec [3.3](https://github.com/reactive-streams/reactive-streams-jvm/tree/v1.0.0#3-subscription-code) says it's the responsibility of `Subscription.request` to put a bound on the recursion, so it is allowed to call `request` from `onNext` as much as you like.